### PR TITLE
Update Patient Dedupe Docs

### PR DESCRIPTION
**Pull request by ksmith94**
This will update the structure, so that there are multiple pages under the Patient Deduplication Architecture docs. 
It will also add new sections and examples to the Matching Rules docs and add examples to the Merge Rules docs.

### CODE REVIEW

File: packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md
Comment (rahul1) [inline packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md:1]: Add a section at the bottom that points to the next document.

"In the next section, we'll talk about the **Ingestion** stage of the pipeline"..."

File: packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+In this guide, we will call each input system that contributes patient data a **source system.** The output of your deduplication pipeline will be called the **target system**.
+
+Most pipelines make a copy of each patient record from the source system into the target system. We will call these copies **source records.** The final, combined patient record will be called the **master record**.
+
+## Pipeline
+
+While deduplication pipelines can take many forms, there three primary operations in every pipeline:
+
+1. **Ingestion:** Copying data from source systems into the target system to create source records.
```
Comment (rahul1) [inline packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md:15]: Link to batch-vs-incremental

File: packages/docs/docs/fhir-datastore/patient-deduplication/conclusion.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Conclusion
```
Comment (rahul1) [inline packages/docs/docs/fhir-datastore/patient-deduplication/conclusion.md:1]: Add this back to the overview doc, and rename the section heading. Doesn't need to be it's own section

**Comment by rahul1** [comment]
We went overboard with breaking this into separate docs. Breakdown mirror the 3 stages of the pipeline: 
1. OVerview
2. Ingestion
3. Matching
4. Merge

**Review by rahul1 — CHANGES_REQUESTED** [review]

**Comment by reshmakh** [comment]
There are broken links in this build



@medplum/docs:build: - On source page path = /docs/fhir-datastore/patient-deduplication/merging:
--
08:06:28.161 | @medplum/docs:build:    -> linking to /docs/fhir-datastore/patient-deduplication/matching-rules#do-not-match-lists (resolved as: /docs/fhir-datastore/patient-deduplication/matching-rules)
08:06:28.161 | @medplum/docs:build:
08:06:28.161 | @medplum/docs:build:     at throwError (/vercel/path0/node_modules/@docusaurus/logger/lib/index.js:76:11)
08:06:28.161 | @medplum/docs:build:     at handleBrokenLinks (/vercel/path0/node_modules/@docusaurus/core/lib/server/brokenLinks.js:153:47)
08:06:28.161 | @medplum/docs:build:     at async buildLocale (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:186:5)
08:06:28.161 | @medplum/docs:build:     at async tryToBuildLocale (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:41:20)
08:06:28.161 | @medplum/docs:build:     at async mapAsyncSequential (/vercel/path0/node_modules/@docusaurus/utils/lib/jsUtils.js:34:24)
08:06:28.161 | @medplum/docs:build:     at async Command.build (/vercel/path0/node_modules/@docusaurus/core/lib/commands/build.js:76:21)
08:06:28.161 | @medplum/docs:build: [INFO] Docusaurus version: 2.4.1

File: packages/docs/docs/fhir-datastore/patient-deduplication/matching-rules.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+| `qualitativeRisk`    | A readable description of how likely the match is.                                            | Very likely           |
+| `method`             | The rule or technique that was used to create the assessment.                                 | last-name             |
+| `code`               | Indicates what is being assessed. When deduping records, it will always be a duplicate match. | duplicate-patient     |
+| `subject`            | The patient or other resource that a potential match is being checked against.                | Patient/homer-simpson |
+| `basis`              | The patient or other resource that could be a potential match for the subject.                | Patient/marge-simpson |
+
+<details><summary>A `RiskAssessment` for a potential duped patient record</summary>
+```js
+{
+  resourceType: RiskAssessment,
```
[the remark this answers is not in the document]
Comment (ksmith94) [reply packages/docs/docs/fhir-datastore/patient-deduplication/matching-rules.md]: These have been moved to a new `ts` file

File: packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md
Comment (ksmith94) [reply packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md:1]: Added

File: packages/docs/docs/fhir-datastore/patient-deduplication/conclusion.md (hunk 2 above)
Comment (ksmith94) [reply packages/docs/docs/fhir-datastore/patient-deduplication/conclusion.md]: This has been moved back to the overview

File: packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md (hunk 1 above)
Comment (ksmith94) [reply packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md]: links are added

File: packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Architecture Overview
+
+## Terminology
```
Comment (reshmakh) [inline packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md:3]: Unfortunately "Terminology" means something specific in FHIR.  Can we change this to "Establishing shared vocabulary", also open to other words but not Terminology as [FHIR Terminology](https://build.fhir.org/terminology-service.html#:~:text=A%20FHIR%20terminology%20service%20is,inherently%20known%20terminologies%20providing%20support.) is something in and of itself.

**Comment by ksmith94** [comment]
This is related to issue #2679

**Comment by rahul1** [comment]
Recorded some additional feedback

File: packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md (hunk 4 above)
Comment (ksmith94) [reply packages/docs/docs/fhir-datastore/patient-deduplication/architecture-overview.md]: This has been changed in the lates commit

**Comment by rahul1** [comment]
@ksmith94 - all the text looks good, just want to change some things with the code. Instead of copy-pasting the code examples into new files, you can just add `start-block` comments to the [original files](https://github.com/medplum/medplum/blob/9df27c2c80e2f72bf58b9a77dcdbff06b539c210/examples/medplum-demo-bots/src/deduplication/find-matching-patients.ts#L1) in medplum-demo-bots.




See an example of how to do that [here](https://github.com/medplum/medplum/blob/9df27c2c80e2f72bf58b9a77dcdbff06b539c210/packages/docs/docs/billing/patient-insurance.md#L13)

**Review by rahul1 — APPROVED** [review]

**Pull request merged**